src/pcapi/core/offerers/repository.py

```diff
@@ -22,3 +25,38 @@ def get_all_offerers_for_user(user: User, filters: dict) -> list[Offerer]:
             query = query.filter(UserOfferer.validationToken.isnot(None))
 
     return query.all()
+
+
+def get_filtered_venues(
+    pro_user_id: int,
+    user_is_admin: bool,
+    active_offerers_only: Optional[bool] = False,
+    offerer_id: Optional[int] = None,
+    validated_offerer: Optional[bool] = None,
+    validated_offerer_for_user: Optional[bool] = None,
+) -> list[Venue]:
+    query = Venue.query.join(Offerer, Offerer.id == Venue.managingOffererId).join(
+        UserOfferer, UserOfferer.offererId == Offerer.id
+    )
+    if not user_is_admin:
+        query = query.filter(UserOfferer.userId == pro_user_id)
+
+    if validated_offerer is not None:
+        if validated_offerer:
+            query = query.filter(Offerer.validationToken.is_(None))
+        else:
+            query = query.filter(Offerer.validationToken.isnot(None))
+    if validated_offerer_for_user is not None:
+        if validated_offerer_for_user:
+            query = query.filter(UserOfferer.validationToken.is_(None))
+        else:
+            query = query.filter(UserOfferer.validationToken.isnot(None))
+
+    if active_offerers_only:
+        query = query.filter(Offerer.isActive.is_(True))
+
+    if offerer_id:
+        query = query.filter(Venue.managingOffererId == offerer_id.persisted)
+
+    venue_list = query.order_by(Venue.name).all()
+    return venue_list
```

Comment:
(détail) la variable n'est pas très utile

```suggestion
    return query.order_by(Venue.name).all()
```